### PR TITLE
chore: Add defer CloseIntegrationTestEnv() to E2E tests

### LINKED ISSUE #1046: Add `defer CloseIntegrationTestEnv` to e2e tests
## 🆕🐥 First Timers Only

This issue is reserved for people who have never contributed to [Hedera](https://hedera.com) or any open source project in general.
We know that creating a pull request (PR) is a major barrier for new contributors.
The goal of this issue and all other issues labeled by [**'Good First Issue'**](https://github.com/issues?q=is%3Aopen+is%3Aissue+org%3Ahashgraph+archived%3Afalse+label%3A%22Good+First+Issue%22+) is to help you make your first contribution to Hedera.

## 👾 Description of the issue

- Certain end-to-end (E2E) tests are not properly closing the testing environment after execution. This can lead to resource leaks and may affect the stability and reliability of subsequent tests.

### Proposed Solution:

- To ensure the testing environment is consistently closed, add a `defer CloseIntegrationTestEnv()` statement in all integration tests (specifically those ending with `_e2e_test.go`). This will automatically close the environment at the end of each test execution and delete the operator account.

 - However, if closing the environment causes a test failure, the environment should remain open to avoid interrupting the test flow. For these cases a follow up issue should be created, containing all the tests' names.

## 📋 Step by step guide to do a contribution

If you have never contributed to an open source project at GitHub, the following step-by-step guide will introduce you to the workflow. More information and concrete samples for shell commands for each step can be found in our [CONTRIBUTING.md](https://github.com/hashgraph/.github/blob/main/CONTRIBUTING.md) file.
A more detailed general documentation of the GitHub PR workflow can be found [here](https://github.com/firstcontributions/first-contributions/blob/master/README.md).

- [ ] **Claim this issue:** Comment below that you are interested in working on the issue
- [ ] **Wait for assignment:** A community member with the given rights will add you as an assignee of the issue
- [ ] **Fork the repository:** You can do that in GitHub (by simply clicking the 'fork' button).
- [ ] **Check out the forked repository**
- [ ] **Create a feature branch** for the issue. We do not have a hard naming definition for branches but it is best practice to prefix the branch name with the issue id.
- [ ] **Solve the issue** in your branch.
- [ ] **Commit your changes:** Here, it is needed to add `sign-off` information to the commit to accept the "Developer Certificate of Origin" (https://developercertificate.org). More details can be found in our [CONTRIBUTING.md](https://github.com/hashgraph/.github/blob/main/CONTRIBUTING.md)
- [ ] **Start a Pull Request (PR)**: We have a pattern for naming pull requests that a GitHub Action checks. We use that pattern to support the creation of automatic release notes.
- [ ] **Check GitHub Actions:** Several GitHub Actions will be triggered automatically for each PR. If a GitHub Action fails and you do not understand the cause of that error do not hesitate to add a comment to the PR and ask the Hedera developer community for support.
- [ ] **Wait for reviews:** Members of the Hedera developer community will review your PR. If a reviewer finds any missing pieces or a problem, he or she will start a discussion with you and describe the next steps for solving the problem.
- [ ] **You did it 🎉:** We will merge the fix in the develop branch. Thanks for being part of the Hedera community as an open-source contributor ❤️

## 🎉 Contribute to Hacktoberfest

Solve this issue as part of the [Hacktoberfest](https://hacktoberfest.digitalocean.com) event and get a chance to receive cool goodies like a T-Shirt. 🎽

## 🤔 Additional Information

If you have any questions, just ask us directly in this issue by adding a comment. You can join our community chat at [Discord](https://hedera.com/discord). A general manual about open-source contributions can be found [here](https://github.com/firstcontributions/first-contributions/blob/master/README.md).

### PR DESCRIPTION
**Pull request by RashRAJ**
**Description**:
This pull request addresses the issue of E2E tests not properly closing the testing environment after execution. It implements a solution to prevent resource leaks and improve the stability and reliability of subsequent tests. The changes include:

1. Adding `defer CloseIntegrationTestEnv()` statements to all integration tests ending with `_e2e_test.go`.

**Related issue(s)**:

Fixes #1046 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by 0xivanov** [comment]
Hey, nice work!
DCO check is failing because the commits are not signed off. Please check out https://github.com/hashgraph/.github/blob/main/CONTRIBUTING.md and see how to set up your GPG keys.
When you've setup the keys you can rebase your commits while signing them.
https://github.com/hashgraph/hedera-sdk-go/pull/1105/checks?check_run_id=31434699874

**Comment by RashRAJ** [comment]
> Hey, nice work! DCO check is failing because the commits are not signed off. Please check out https://github.com/hashgraph/.github/blob/main/CONTRIBUTING.md and see how to set up your GPG keys. When you've setup the keys you can rebase your commits while signing them. https://github.com/hashgraph/hedera-sdk-go/pull/1105/checks?check_run_id=31434699874

Hi, @0xivanov  Fixed the verification issue. Thanks for the links!

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/hashgraph/hedera-sdk-go/v2
 
-go 1.20
+go 1.21
+
+toolchain go1.23.2
```
Comment (0xivanov) [inline go.mod:5]: why change the go version?

Comment (RashRAJ) [reply go.mod:5]: The version change was made during testing, but I can revert to the previous version if needed.

Comment (0xivanov) [reply go.mod:5]: Yes, please revert it.

**Comment by 0xivanov** [comment]
The test build is failing. 
Please refer to this: https://github.com/hashgraph/hedera-sdk-go/pull/1097#issuecomment-2402388255

**Comment by RashRAJ** [comment]
> The test build is failing. Please refer to this: [#1097 (comment)](https://github.com/hashgraph/hedera-sdk-go/pull/1097#issuecomment-2402388255)

Hi @0xivanov I have pushed the fix

**Comment by 0xivanov** [comment]
`TestIntegrationAccountIDCanPopulateAccountAliasEvmAddressWithNoMirror` test is failing.

**Comment by RashRAJ** [comment]
> `TestIntegrationAccountIDCanPopulateAccountAliasEvmAddressWithNoMirror` test is failing.

Adding `defer CloseIntegrationTestEnv()` causes this test failure. It affects both `TestIntegrationAccountIDCanPopulateAccountAliasEvmAddressWithMirrorAndNoEvmAddress` and `TestIntegrationAccountIDCanPopulateAccountAliasEvmAddressWithNoMirror.` Removing it fixed the test failure

**Review by 0xivanov — APPROVED** [review]
LGTM

**Pull request merged**